### PR TITLE
updated C microphone demo to support windows and mac

### PR DESCRIPTION
**Pull request by ksyeo1010**
Updated C microphone demo to use `miniaudio`.  Added `miniaudio` as submodule and added support for Windows and Mac. Updated readme to reflect changes.

### CODE REVIEW

File: demo/c/README.md [hunk 1]
```diff
@@ -1,66 +1,102 @@
...
+gcc -std=c99 -O3 -o demo/c/rhino_demo_mic -I include demo/c/rhino_demo_mic.c -ldl -lpthread -lm
 ```
 
-Running the executable without any commandline arguments prints the usage info to the console as below
+### Windows
 
 ```console
-$ demo/c/rhino_demo_mic
-usage : demo/c/rhino_demo_mic library_path model_path context_path input_audio_device
+gcc -std=c99 -O3 -o demo/c/rhino_demo_mic -I include demo/c/rhino_demo_mic.c
```
Comment (kenarsa) [inline demo/c/README.md:27]: doesn't need `-lm`?

File: demo/c/README.md [hunk 2]
```diff
@@ -1,66 +1,102 @@
...
-within the context of a smart lighting system. For example you can say "turn on the lights"
+Running the executable without any commandline arguments prints the usage info to the console:
 
 ```console
-demo/c/rhino_demo_mic \
-lib/linux/x86_64/libpv_rhino.so \
-lib/common/rhino_params.pv \
-resources/contexts/linux/smart_lighting_linux.rhn \
-plughw:CARD=AK5371
+./demo/c/rhino_demo_mic
```
Comment (kenarsa) [inline demo/c/README.md:35]: windows?

File: demo/c/README.md [hunk 3]
```diff
@@ -1,66 +1,102 @@
...
+usage : ./demo/c/rhino_demo_mic library_path model_path context_path input_audio_device
+        ./demo/c/rhino_demo_mic --show_audio_devices
 ```
 
-The following achieves the same on a Raspberry Pi 4
+To show the available audio input devices, on Linux, macOS, Raspberry Pi run:
```
Comment (kenarsa) [inline demo/c/README.md:40]: windows?

**Review by kenarsa — CHANGES_REQUESTED** [review]

File: demo/c/README.md (hunk 1 above)
Comment (ksyeo1010) [reply demo/c/README.md:26]: yes, doesn't need any linking

**Review by kenarsa — APPROVED** [review]

**Pull request merged**